Dev/rackscale scheduler

hunhoffe opened the pull request:
The integration test for dcm that I've written will fail on the runner. This is because you'll need to follow the directions below to get it working on a new machine:
- Follow the instruction [here](https://github.com/hunhoffe/nrk-dcm-scheduler/tree/scheduler) to install the right version of java and maven. What exactly you have to do to make it work correctly depends on if you have other version of java and/or maven already on your system
- In your local nrk git repo, checkout the ```dev/rackscale-scheduler``` branch
- ```cd``` into ```nrk/nrk-dcm-scheduler```
- Run the ```setup.sh``` script
- To run the integration test with the scheduler, run: ```cargo test --test integration-test -- dcm```

gz commented on lib/rpc/src/transport/smoltcp.rs:
```diff
@@ -19,7 +20,7 @@ const RX_BUF_LEN: usize = 8192;
 
 pub struct TCPTransport<'a> {
-    iface: RefCell<Interface<'a, DevQueuePhy>>,
+    iface: Arc<RefCell<Interface<'a, DevQueuePhy>>>,
```
This isn't for this PR but longer-term: we should think about what to do with the device Interface since RefCell is only for a single core.
cc @prestonsn this might be a good place to start looking into the NR+device interface model without having to implement an NVMe device first but use a NIC device instead

hunhoffe commented on .gitmodules:
```diff
@@ -2,3 +2,7 @@
 	path = lib/node-replication
 	url = git@github.com:vmware/node-replication.git
 	branch = mut_scan_vec
+[submodule "nrk-dcm-scheduler"]
```
@gz Now that I can use github releases, is there a reason to leave this as a submodule? I couldn't think of one, so if you can't either, I will remove it as a submodule.

gz replied:
Yes lets drop the submodule again, but we should have some way to override the jar with a local custom version (maybe this can be part of run.py though)...
Maybe run.py can also contain the build commanda and we just point it to the right directory in this case?

gz replied:
just fyi removing a submodule again
is quite the hassle but these steps tend to work: https://nrkernel.systems/book/development/Configuration.html#removing-a-submodule-in-the-repository

hunhoffe replied:
Thank you, I should have looked it up first. Apologies!

hunhoffe replied:
Is there anything else I should do to fix things further on the runners, since I didn't remove the submodule gracefully?

hunhoffe commented:
I added github CI and automatic releases (upon commits and PRs to main) in the nrk-dcm-scheduler github repo. I now changed the run.py script to fetch a given release (unless a jar path is specified on the command line, for the case of local development). 

This means that to run DCM tests, runners only need to have java installed (no maven, etc.)
